packages/msw-addon/src/mswDecorator.js

```diff
@@ -1,23 +1,41 @@
-import { makeDecorator } from '@storybook/addons';
 import { setupWorker } from 'msw';
+import { setupServer } from 'msw/node';
 
 let worker;
 
+const IS_BROWSER = typeof global.process === 'undefined'
+
 export function initializeWorker(options) {
-  if (typeof global.process === 'undefined') {
+  if (IS_BROWSER) {
     worker = setupWorker();
     worker.start(options);
+  } else {
+    worker = setupServer();
```

Comment:
Assigning a Node.js API to the `worker` variable is confusing. 

It looks like the `worker` exists for two reasons:

1. To throw an error if you attempt to call `getWorker()` without initializing the plugin.
2. To return the reference to the API.

We can implement the first point by having a boolean representing whether `initializeWorker` has been called.
The second point is much harder to tackle, because the introduction of `setupServer` makes the return type of `initiailizeWorker` ambiguous. Perhaps, we could return `null` in the case of Node.js.

Reply:
Btw I noticed that in other PR you decided to initialize the worker when running the decorator, and I think it's not ideal, given that the worker would be initialized X times, where X is the amount of stories, in comparison to just once, in the preview.js file. Same would apply for the addon in node.js